inference-engine/tests/unit/vpu/middleend_tests/passes_tests/reshape_before_conv.cpp

```diff
@@ -0,0 +1,550 @@
+    }
+
+    void InitPipeline() {
+        _pipeline = PassSet();
+        _pipeline.addPass(passManager->dumpModel("before-convert-shape-notation"));
+        _pipeline.addPass(passManager->reshapeBeforeConvTiling());
+        _pipeline.addPass(passManager->dumpModel("after-convert-shape-notation"));
+    }
+
+    Data InitInputData(const DimValues& dims) {
+        const auto inDesc = DataDesc(DataType::FP16, DimsOrder::NCHW, dims);
+        return _model->addInputData("Input", inDesc);
+    }
+
+    Data InitOutputData(const DimValues& dims) {
+        const auto outDesc = DataDesc(DataType::FP16, DimsOrder::NCHW, dims);
+        return _model->addOutputData("Output", outDesc);
+    }
+
+    Data InitNewData(const DimValues& dims, const std::string& name) {
+        const auto newDesc = DataDesc(DataType::FP16, DimsOrder::NCHW, dims);
+        return _model->addNewData(name, newDesc);
+    }
+
+    void InitConvStage(const Data& input, const Data& output, int kernel_x = 1, int kernel_y = 1) {
+        auto convLayer = std::make_shared < ie::ConvolutionLayer > (ie::LayerParams {
+            "TestConvolution",
+            "StubConv",
+            ie::Precision::FP16
+        });
+        convLayer->_kernel_x = kernel_x;
+        convLayer->_kernel_y = kernel_y;
+        convLayer->_stride_x = 1;
+        convLayer->_stride_y = 1;
+
+        Stage stage = stageBuilder->addConvolutionStage(_model, "TestConvolution", convLayer, input, output,
+                InitNewData(DimValues{ {Dim::N, output->desc().dim(Dim::C)}, {Dim::C, input->desc().dim(Dim::C)}, {Dim::H, kernel_y}, {Dim::W, kernel_x} },
+                "TestConvolution@weights@conv"),
+                _model->addFakeData(), _model->addFakeData());
+
+        stage->attrs().set<int>("kernelSizeX", kernel_x);
+        stage->attrs().set<int>("kernelSizeY", kernel_y);
+        stage->attrs().set<int>("kernelStrideX", 1);
+        stage->attrs().set<int>("kernelStrideY", 1);
+        stage->attrs().set<bool>("tryHW", true);
+    }
+
+    void Validate() {
+        const auto datas = _model->datas();
+        const auto stages = _model->getStages();
+        ASSERT_EQ(datas.size(), pattern_datas.size());
+        ASSERT_EQ(stages.size(), pattern_stages.size());
+
+        auto dataIter = datas.begin();
+        for (auto patternIter = pattern_datas.begin(); patternIter != pattern_datas.end();
+                ++dataIter,
+                ++patternIter) {
+            ASSERT_EQ((*dataIter)->desc(), (*patternIter)->desc());
+            ASSERT_EQ((*dataIter)->name(), (*patternIter)->name());
+        }
+
+        auto stageIter = stages.begin();
+        for (auto patternIter = pattern_stages.begin(); patternIter != pattern_stages.end();
+                ++stageIter,
+                ++patternIter) {
+            ASSERT_EQ((*stageIter)->type(), *patternIter);
+        }
+    }
+
+protected:
+    PassSet _pipeline;
+    Model _model;
+    std::vector<Data>pattern_datas;
```

Comment:
Space between type and variable name?

Does it even compile?

Reply:
Are you sure you need these vectors. `Model` already contains data objects and stages. Could you just reuse this knowledge?

Reply:
solved